src/signed.rs

```diff
@@ -192,9 +193,9 @@ impl<'a, T: Signable + Clone + Index, KB: KeyBox> Clone for Signed<'a, T, KB> {
 
 impl<'a, T: Signable + Index, KB: KeyBox> Signed<'a, T, KB> {
     /// Create a signed object from a signable. The index of `signable` must match the index of the `key_box`.
-    pub fn sign(signable: T, key_box: &'a KB) -> Self {
+    pub async fn sign(signable: T, key_box: &'a KB) -> Signed<'a, T, KB> {
```

Comment:
Huh, `Self` doesn't work with `async`?

Reply:
The compiler complains when `Self` has lifetimes